package.json

```diff
@@ -78,7 +79,8 @@
     "@bugsnag/js": "^7.16.2",
     "@types/react-dom": "^17.0.9",
     "graphql": "16.0.0",
-    "lerna": "^4.0.0"
+    "lerna": "^4.0.0",
+    "papaparse": "^5.3.2"
```

Comment:
One the other hand serializing an object to csv is probably less than 10 lines of code and you could remove an external dependency. 

(go influenced: https://www.youtube.com/watch?v=PAAkCSZUG1c&t=568s )

Reply:
completely agree - always keen to reduce dependencies and if we can implement easily then we should do it

Reply:
Just a caveat -- there's all kinds of gotchas when parsing CSV -- escaping characters, escaping delimiters, handling special characters, non-standard text encodings, etc... so I'd be careful about implementing our own solution when this library does a good job with pretty much anything you can throw at it:

As they say in papaparse docs:
<img width="782" alt="Screen Shot 2022-04-14 at 3 46 17 PM" src="https://user-images.githubusercontent.com/5456533/163310081-779801a5-140d-4fe5-af7b-fca9ebb71f1f.png">

😆

Reply:
...for the unparse alone - several hundred lines of code. I was thinking to keep papa in place, even though we don't currently need parsing, only unparsing! As a general principle though.. I'd like to keep the external packages to a minimum.

I think excel handling is probably more useful tbh. Being able to format nicely is good 😉  
We have import facilities in mSupply using excel rather than csv.. however, csv is probably more aligned with an open source, keep-it-free product though.. so maybe we should just support that? ditch the commercial option 🤷

Reply:
> As a general principle though.. I'd like to keep the external packages to a minimum.

For sure! I just don't like having to anticipate and write guards for a million different edge cases when someone else has already solved the problem. (I'm actually only bothering to bring this up at all cos I learnt it the hard way -- I recently wrote a csv exporter for Conforma, which worked for the most part, but escaping quotes inside strings properly and other hassles became such headache that I was glad to just pass the problem over to papaparse -- so it's a fresh wound 😄)

> I think excel handling is probably more useful tbh. Being able to format nicely is good 😉

I sort of agree, but personally I'd rather use Google Sheets than Excel if I can help it. ;) (But if you export to Excel, then Google Sheets can easily import that and preserve formatting/formulas, etc anyway). But even if you do allow export to Excel, I think it should always have a CSV option as well -- not everyone want the hassle of dealing with Excel.

Reply:
Yes only talking about serialization, de-serialization is much harder. I have no objecting in using the lib just saying that we could do without, without doing a terrible bad thing :-)

Isn't excel pretty good in importing csv? (only know that libreoffice has no problem with it)

Reply:
> Isn't excel pretty good in importing csv

You have to be vigilant about switching off the auto-formatting options, otherwise Excel is notorious for converting things to dates and numbers that should be kept as text (just [ask the geneticists](https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates)!)